app/upgrades/v16/upgrades.go

```diff
@@ -80,7 +82,44 @@ func CreateUpgradeHandler(
 		defaultConcentratedLiquidityParams.IsPermissionlessPoolCreationEnabled = IsPermissionlessPoolCreationEnabledCL
 		keepers.ConcentratedLiquidityKeeper.SetParams(ctx, defaultConcentratedLiquidityParams)
 
-		if err := createCanonicalConcentratedLiquidityPoolAndMigrationLink(ctx, DaiOsmoPoolId, DesiredDenom0, keepers); err != nil {
+		// Create a concentrated liquidity pool for DAI/OSMO.
+		// Link the DAI/OSMO balancer pool to the cl pool.
+		clPool, err := createCanonicalConcentratedLiquidityPoolAndMigrationLink(ctx, DaiOsmoPoolId, DesiredDenom0, keepers)
+		if err != nil {
+			return nil, err
+		}
+		clPoolId := clPool.GetId()
+		clPoolDenom := cltypes.GetConcentratedLockupDenomFromPoolId(clPoolId)
+
+		// Create a position to initialize the balancerPool.
+
+		// Get the address of the balancerPool.
+		balancerPool, err := keepers.PoolManagerKeeper.GetPool(ctx, DaiOsmoPoolId)
+		if err != nil {
+			return nil, err
+		}
+
+		// Retrieve the spot price of the OSMO/DAI pool at the time of the upgrade.
+		// This is used to initialize the balancerPool's spot price.
+		spotPrice, err := balancerPool.SpotPrice(ctx, DAIIBCDenom, DesiredDenom0)
+		if err != nil {
+			return nil, err
+		}
+		quoteAssetAmount := spotPrice.Mul(sdk.NewDec(1000000)).TruncateInt()
+
+		coins := sdk.NewCoins(sdk.NewCoin(DesiredDenom0, sdk.NewInt(1000000)), sdk.NewCoin(DAIIBCDenom, quoteAssetAmount))
+		_, _, _, _, _, err = createFullRangePositionNoSend(ctx, clPoolId, balancerPool.GetAddress(), coins, *keepers.ConcentratedLiquidityKeeper)
```

Comment:
So we add liquidity with tokens that do not exist?

Is this appropriate? Can there be any unintended consequences of this?

E.g. how fee or incentive distributions is going to be affected?

Reply:
Yeah, I was torn on this, but after thinking about it I couldn't really think of any edge cases that this would effect (the pool acc would never actually redeem this position / collect fees / collect incentives). 

The only other alternative I could think of is sending funds directly from the balancer osmo/dai pool which also doesn't feel like a great solution. Happy to discuss further options though

Reply:
Got on call, decided to potentially try and run a swap from the community pool to get some DAI and make a real position this way

Reply:
Don't quite understand how this could work, would you be able to elaborate more on why creating full range positions without the actual coins would make sense?

Reply:
It was initially just to make a spot price, but we decided to go with creating a real position with actual funds using the community pool instead